internal/cmd/beta/security-group/rule/create/create.go

```diff
@@ -0,0 +1,268 @@
+			),
+			examples.NewExample(
+				`Create a security group rule for security group with ID "xxx" with direction "ingress" and protocol number 1 `,
+				`$ stackit beta security-group rule create --security-group-id xxx --direction ingress --protocol-number 1`,
+			),
+		),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			projectLabel, err := projectname.GetProjectName(ctx, p, cmd)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get project name: %v", err)
+				projectLabel = model.ProjectId
+			}
+
+			securityGroupLabel, err := iaasUtils.GetSecurityGroupName(ctx, apiClient, model.ProjectId, model.SecurityGroupId)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get security group name: %v", err)
+				securityGroupLabel = model.SecurityGroupId
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to create a security group rule for security group %q for project %q?", securityGroupLabel, projectLabel)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create security group rule : %w", err)
+			}
+
+			return outputResult(p, model, projectLabel, securityGroupLabel, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), securityGroupIdFlag, "The security group ID")
+	cmd.Flags().String(directionFlag, "", "The direction of the traffic which the rule should match. The possible values are: `ingress`, `egress`")
+	cmd.Flags().String(descriptionFlag, "", "The rule description")
+	cmd.Flags().String(etherTypeFlag, "", "The ethertype which the rule should match")
+	cmd.Flags().Int64(icmpParameterCodeFlag, 0, "ICMP code. Can be set if the protocol is ICMP")
+	cmd.Flags().Int64(icmpParameterTypeFlag, 0, "ICMP type. Can be set if the protocol is ICMP")
+	cmd.Flags().String(ipRangeFlag, "", "The remote IP range which the rule should match")
+	cmd.Flags().Int64(portRangeMaxFlag, 0, "The maximum port number. Should be greater or equal to the minimum. This should only be provided if the protocol is not ICMP")
+	cmd.Flags().Int64(portRangeMinFlag, 0, "The minimum port number. Should be less or equal to the maximum. This should only be provided if the protocol is not ICMP")
+	cmd.Flags().Var(flags.UUIDFlag(), remoteSecurityGroupIdFlag, "The remote security group which the rule should match")
+	cmd.Flags().Int64(protocolNumberFlag, 0, "The protocol number which the rule should match. If a protocol is to be defined, either `name` or `number` must be provided")
```

Comment:
Also when referring to flags I think we should use the complete flag name, like `protocol-name` and `protocol-number` instead of just `name` and `number`

Reply:
`"If a protocol is to be defined" ` means if you want to specifically define a protocol by giving `--protocol-name` or `--protocol-number`

Reply:
Ah ok

Reply:
Not sure if that part is needed then, but if doesn't make the description too long I would say it's fine